pyairtable/api/abstract.py

```diff
@@ -16,25 +16,30 @@
 
 class ApiAbstract(metaclass=abc.ABCMeta):
     VERSION = "v0"
-    API_BASE_URL = "https://api.airtable.com/"
     API_LIMIT = 1.0 / 5  # 5 per second
-    API_URL = posixpath.join(API_BASE_URL, VERSION)
     MAX_RECORDS_PER_REQUEST = 10
 
     session: Session
+    endpoint_url: str
     tiemout: TimeoutTuple
 
     def __init__(
         self,
         api_key: str,
         timeout: Optional[TimeoutTuple] = None,
         retry_strategy: Optional[Retry] = None,
+        endpoint_url: str = "https://api.airtable.com",
     ):
         if not retry_strategy:
             self.session = Session()
         else:
             self.session = _RetryingSession(retry_strategy)
 
+        if not endpoint_url.endswith(self.VERSION):
+            self.endpoint_url = posixpath.join(endpoint_url, self.VERSION)
+        else:
+            self.endpoint_url = endpoint_url
+        print(self.endpoint_url)
```

Comment:
Please remove the `print()` – I'm guessing that was probably just leftover from debugging? Otherwise lgtm.

(FWIW, I don't think we need to support two different URL formats, it's probably fine to just use the provided value as-is.)

Reply:
Oh, one more request: can you add a mention of this new parameter in the API documentation?

Reply:
> Please remove the `print()` – I'm guessing that was probably just leftover from debugging? Otherwise lgtm.

Whoops https://github.com/gtalarico/pyairtable/pull/243/commits/66bc47941aaa1ed533ed373f2fbebef28e4297af

> Oh, one more request: can you add a mention of this new parameter in the API documentation?

Copied and adjusted from airtable.js
https://github.com/gtalarico/pyairtable/pull/243/commits/14e5d49fd8682b0db099122c611099b299368e06